crypto/crypto/src/unit_tests/multi_ed25519_test.rs

```diff
@@ -18,9 +18,11 @@ use rand::{rngs::StdRng, SeedableRng};
 static MESSAGE_HASH: Lazy<HashValue> = Lazy::new(|| HashValue::from_sha3_256(b"Test Message"));
 
 // Helper function to generate N key pairs.
-fn generate_key_pairs(n: usize) -> Vec<(Ed25519PrivateKey, Ed25519PublicKey)> {
+fn generate_keys(n: usize) -> Vec<Ed25519PrivateKey> {
     let mut rng = StdRng::from_seed(TEST_SEED);
-    (0..n).map(|_| compat::generate_keypair(&mut rng)).collect()
+    (0..n)
+        .map(|_| Ed25519PrivateKey::generate(&mut rng))
```

Comment:
+1, we shouldn't use `generate_for_testing` here as many keys are derived from the same "starting" seed, but the seed should be different between them.

Reply:
note this is similar to the AC above.